File: android/app/src/main/java/com/expensetracker/ui/screens/AuthScreens.kt
package com.expensetracker.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.expensetracker.ui.theme.*
import com.expensetracker.ui.components.*

data class OnboardingPage(
    val title: String,
    val subtitle: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
)

@Composable
fun WelcomeScreen(
    onGetStarted: () -> Unit = {}
) {
    var currentPage by remember { mutableStateOf(0) }
    
    val pages = listOf(
        OnboardingPage(
            title = "Track Your Expenses",
            subtitle = "Keep track of your family's income and expenses with our easy-to-use interface",
            icon = Icons.Default.PieChart
        ),
        OnboardingPage(
            title = "Set Savings Goals",
            subtitle = "Create and monitor your savings goals to achieve your financial dreams",
            icon = Icons.Default.TrendingUp
        ),
        OnboardingPage(
            title = "Family Financial Management",
            subtitle = "Manage your family's finances together with multiple user accounts and permissions",
            icon = Icons.Default.Group
        )
    )
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp)
            .verticalScroll(rememberScrollState()),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(40.dp))
        
        // Illustration placeholder
        Box(
            modifier = Modifier
                .size(200.dp)
                .background(
                    PrimaryBlue.copy(alpha = 0.1f),
                    CircleShape
                ),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = pages[currentPage].icon,
                contentDescription = null,
                tint = PrimaryBlue,
                modifier = Modifier.size(120.dp)
            )
        }
        
        Spacer(modifier = Modifier.height(40.dp))
        
        // Title and subtitle
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Text(
                text = pages[currentPage].title,
                style = MaterialTheme.typography.displayMedium,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center,
                color = MaterialTheme.colorScheme.onSurface
            )
            
            Text(
                text = pages[currentPage].subtitle,
                style = MaterialTheme.typography.bodyLarge,
                textAlign = TextAlign.Center,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f),
                modifier = Modifier.padding(horizontal = 16.dp)
            )
        }
        
        Spacer(modifier = Modifier.weight(1f))
        
        // Page indicators
        Row(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            modifier = Modifier.padding(bottom = 32.dp)
        ) {
            repeat(pages.size) { index ->
                Box(
                    modifier = Modifier
                        .size(8.dp)
                        .background(
                            if (index == currentPage) PrimaryBlue else CardBorder,
                            CircleShape
                        )
                )
            }
        }
        
        // Navigation buttons
        Column(
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            if (currentPage < pages.size - 1) {
                PrimaryButton(
                    text = "Next",
                    onClick = { currentPage++ },
                    modifier = Modifier.fillMaxWidth()
                )
            } else {
                PrimaryButton(
                    text = "Get Started",
                    onClick = onGetStarted,
                    modifier = Modifier.fillMaxWidth()
                )
            }
            
            if (currentPage > 0) {
                TextButton(
                    onClick = { currentPage-- },
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text(
                        "Previous",
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                    )
                }
            }
        }
    }
}

@Composable
fun AuthSelectionScreen(
    onEmailAuth: () -> Unit = {},
    onGoogleAuth: () -> Unit = {}
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(60.dp))
        
        // App logo
        Icon(
            imageVector = Icons.Default.AccountBalance,
            contentDescription = null,
            tint = PrimaryBlue,
            modifier = Modifier.size(120.dp)
        )
        
        Spacer(modifier = Modifier.height(32.dp))
        
        // Title and subtitle
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Text(
                text = "Family Finance Tracker",
                style = MaterialTheme.typography.displayMedium,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center,
                color = MaterialTheme.colorScheme.onSurface
            )
            
            Text(
                text = "Manage your family's finances together",
                style = MaterialTheme.typography.bodyLarge,
                textAlign = TextAlign.Center,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
            )
        }
        
        Spacer(modifier = Modifier.weight(1f))
        
        // Auth buttons
        Column(
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Google Sign In
            SecondaryButton(
                text = "Continue with Google",
                onClick = onGoogleAuth,
                modifier = Modifier.fillMaxWidth()
            )
            
            // Email Sign In
            PrimaryButton(
                text = "Continue with Email",
                onClick = onEmailAuth,
                modifier = Modifier.fillMaxWidth()
            )
            
            // Terms and Privacy
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier.padding(top = 16.dp)
            ) {
                Text(
                    text = "By continuing, you agree to our",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                )
                
                Row(
                    horizontalArrangement = Arrangement.spacedBy(4.dp)
                ) {
                    TextButton(onClick = { /* Terms */ }) {
                        Text(
                            "Terms of Service",
                            style = MaterialTheme.typography.bodySmall,
                            color = PrimaryBlue
                        )
                    }
                    
                    Text(
                        "and",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                    )
                    
                    TextButton(onClick = { /* Privacy */ }) {
                        Text(
                            "Privacy Policy",
                            style = MaterialTheme.typography.bodySmall,
                            color = PrimaryBlue
                        )
                    }
                }
            }
        }
    }
}

@Composable
fun EnhancedLoginScreen(
    onNavigateToDashboard: () -> Unit = {}
) {
    var email by remember { mutableStateOf("") }
    var password by remember { mutableStateOf("") }
    var isLoading by remember { mutableStateOf(false) }
    var isSignUp by remember { mutableStateOf(false) }
    
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp)
            .verticalScroll(rememberScrollState()),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(60.dp))
        
        // Logo
        Icon(
            imageVector = Icons.Default.AccountBalance,
            contentDescription = null,
            tint = PrimaryBlue,
            modifier = Modifier.size(80.dp)
        )
        
        Spacer(modifier = Modifier.height(32.dp))
        
        // Title and subtitle
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Text(
                text = if (isSignUp) "Create Account" else "Welcome Back",
                style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Bold,
                color = MaterialTheme.colorScheme.onSurface
            )
            
            Text(
                text = if (isSignUp) "Join Family Finance Tracker" else "Sign in to your account",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
            )
        }
        
        Spacer(modifier = Modifier.height(40.dp))
        
        // Form fields
        Column(
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            FormTextField(
                label = "Email",
                value = email,
                onValueChange = { email = it },
                placeholder = "Enter your email",
                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Email)
            )
            
            Column(
                verticalArrangement = Arrangement.spacedBy(4.dp)
            ) {
                Text(
                    text = "Password",
                    style = MaterialTheme.typography.labelLarge,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                OutlinedTextField(
                    value = password,
                    onValueChange = { password = it },
                    placeholder = { Text("Enter your password") },
                    visualTransformation = PasswordVisualTransformation(),
                    keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedBorderColor = PrimaryBlue,
                        unfocusedBorderColor = CardBorder
                    )
                )
            }
            
            Spacer(modifier = Modifier.height(8.dp))
            
            PrimaryButton(
                text = if (isLoading) "" else if (isSignUp) "Create Account" else "Sign In",
                onClick = {
                    isLoading = true
                    // TODO: Implement actual auth
                    kotlinx.coroutines.GlobalScope.launch {
                        kotlinx.coroutines.delay(1000)
                        onNavigateToDashboard()
                    }
                },
                modifier = Modifier.fillMaxWidth()
            )
            
            TextButton(
                onClick = { isSignUp = !isSignUp },
                modifier = Modifier.fillMaxWidth()
            ) {
                Text(
                    if (isSignUp) "Already have an account? Sign In" else "Don't have an account? Sign Up",
                    color = PrimaryBlue
                )
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
fun WelcomeScreenPreview() {
    ExpenseTrackerTheme {
        WelcomeScreen()
    }
}

@Preview(showBackground = true)
@Composable
fun AuthSelectionScreenPreview() {
    ExpenseTrackerTheme {
        AuthSelectionScreen()
    }
}

@Preview(showBackground = true)
@Composable
fun EnhancedLoginScreenPreview() {
    ExpenseTrackerTheme {
        EnhancedLoginScreen()
    }
}

File: ios/ExpenseTracker/Screens/AuthScreens.swift
import SwiftUI

struct WelcomeView: View {
    @State private var currentPage = 0
    private let pages = [
        OnboardingPage(
            title: "Track Your Expenses",
            subtitle: "Keep track of your family's income and expenses with our easy-to-use interface",
            imageName: "chart.pie"
        ),
        OnboardingPage(
            title: "Set Savings Goals",
            subtitle: "Create and monitor your savings goals to achieve your financial dreams",
            imageName: "target"
        ),
        OnboardingPage(
            title: "Family Financial Management",
            subtitle: "Manage your family's finances together with multiple user accounts and permissions",
            imageName: "person.3"
        )
    ]
    
    var body: some View {
        VStack(spacing: Theme.Spacing.xxxl) {
            Spacer()
            
            // Page indicator placeholder image
            Image(systemName: pages[currentPage].imageName)
                .font(.system(size: 120))
                .foregroundColor(Theme.Colors.primary)
                .frame(height: 200)
            
            VStack(spacing: Theme.Spacing.lg) {
                Text(pages[currentPage].title)
                    .font(Theme.Typography.titleXL)
                    .foregroundColor(Theme.Colors.onSurface)
                    .multilineTextAlignment(.center)
                
                Text(pages[currentPage].subtitle)
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, Theme.Spacing.xl)
            }
            
            Spacer()
            
            VStack(spacing: Theme.Spacing.lg) {
                // Page indicators
                HStack(spacing: Theme.Spacing.sm) {
                    ForEach(0..<pages.count, id: \.self) { index in
                        Circle()
                            .fill(index == currentPage ? Theme.Colors.primary : Theme.Colors.cardBorder)
                            .frame(width: 8, height: 8)
                    }
                }
                
                // Navigation buttons
                if currentPage < pages.count - 1 {
                    PrimaryButton(title: "Next") {
                        withAnimation(.easeInOut(duration: 0.3)) {
                            currentPage += 1
                        }
                    }
                } else {
                    NavigationLink(destination: AuthSelectionView()) {
                        Text("Get Started")
                            .font(Theme.Typography.label)
                            .foregroundColor(Theme.Colors.onPrimary)
                            .padding(.horizontal, Theme.Spacing.lg)
                            .padding(.vertical, Theme.Spacing.md)
                            .background(
                                RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                                    .fill(Theme.Colors.primary)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())
                }
                
                if currentPage > 0 {
                    Button("Previous") {
                        withAnimation(.easeInOut(duration: 0.3)) {
                            currentPage -= 1
                        }
                    }
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                }
            }
        }
        .padding(Theme.Spacing.xl)
        .background(Theme.Colors.surface)
        .navigationBarHidden(true)
    }
}

struct OnboardingPage {
    let title: String
    let subtitle: String
    let imageName: String
}

struct AuthSelectionView: View {
    var body: some View {
        VStack(spacing: Theme.Spacing.xxxl) {
            Spacer()
            
            // App logo/illustration placeholder
            Image(systemName: "dollarsign.circle.fill")
                .font(.system(size: 120))
                .foregroundColor(Theme.Colors.primary)
            
            VStack(spacing: Theme.Spacing.lg) {
                Text("Family Finance Tracker")
                    .font(Theme.Typography.titleXL)
                    .foregroundColor(Theme.Colors.onSurface)
                    .multilineTextAlignment(.center)
                
                Text("Manage your family's finances together")
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                    .multilineTextAlignment(.center)
            }
            
            Spacer()
            
            VStack(spacing: Theme.Spacing.lg) {
                // Google Sign In Button
                Button(action: {
                    // Handle Google sign in
                }) {
                    HStack(spacing: Theme.Spacing.md) {
                        Image(systemName: "globe")
                            .foregroundColor(Theme.Colors.onSurface)
                        
                        Text("Continue with Google")
                            .font(Theme.Typography.label)
                            .foregroundColor(Theme.Colors.onSurface)
                    }
                    .padding(.horizontal, Theme.Spacing.lg)
                    .padding(.vertical, Theme.Spacing.md)
                    .frame(maxWidth: .infinity)
                    .background(
                        RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                            .fill(Theme.Colors.card)
                            .overlay(
                                RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                                    .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                            )
                    )
                }
                
                // Email Sign In/Up
                NavigationLink(destination: LoginView()) {
                    HStack(spacing: Theme.Spacing.md) {
                        Image(systemName: "envelope")
                            .foregroundColor(Theme.Colors.onPrimary)
                        
                        Text("Continue with Email")
                            .font(Theme.Typography.label)
                            .foregroundColor(Theme.Colors.onPrimary)
                    }
                    .padding(.horizontal, Theme.Spacing.lg)
                    .padding(.vertical, Theme.Spacing.md)
                    .frame(maxWidth: .infinity)
                    .background(
                        RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                            .fill(Theme.Colors.primary)
                    )
                }
                .buttonStyle(PlainButtonStyle())
                
                // Terms and Privacy
                VStack(spacing: Theme.Spacing.xs) {
                    Text("By continuing, you agree to our")
                        .font(Theme.Typography.caption)
                        .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                    
                    HStack(spacing: 4) {
                        Button("Terms of Service") {
                            // Handle terms
                        }
                        .font(Theme.Typography.caption)
                        .foregroundColor(Theme.Colors.primary)
                        
                        Text("and")
                            .font(Theme.Typography.caption)
                            .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                        
                        Button("Privacy Policy") {
                            // Handle privacy
                        }
                        .font(Theme.Typography.caption)
                        .foregroundColor(Theme.Colors.primary)
                    }
                }
            }
        }
        .padding(Theme.Spacing.xl)
        .background(Theme.Colors.surface)
        .navigationBarHidden(true)
    }
}

struct EnhancedLoginView: View {
    @EnvironmentObject var authManager: AuthManager
    @State private var email = ""
    @State private var password = ""
    @State private var isSignUp = false
    
    var body: some View {
        VStack(spacing: Theme.Spacing.xl) {
            Spacer()
            
            // Logo
            Image(systemName: "dollarsign.circle.fill")
                .font(.system(size: 80))
                .foregroundColor(Theme.Colors.primary)
            
            VStack(spacing: Theme.Spacing.lg) {
                Text(isSignUp ? "Create Account" : "Welcome Back")
                    .font(Theme.Typography.titleL)
                    .foregroundColor(Theme.Colors.onSurface)
                
                Text(isSignUp ? "Join Family Finance Tracker" : "Sign in to your account")
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
            }
            
            VStack(spacing: Theme.Spacing.lg) {
                FormTextField(
                    label: "Email",
                    text: $email,
                    placeholder: "Enter your email"
                )
                .keyboardType(.emailAddress)
                .autocapitalization(.none)
                
                VStack(alignment: .leading, spacing: Theme.Spacing.xs) {
                    Text("Password")
                        .font(Theme.Typography.label)
                        .foregroundColor(Theme.Colors.onSurface)
                    
                    SecureField("Enter your password", text: $password)
                        .font(Theme.Typography.body)
                        .padding(Theme.Spacing.md)
                        .background(
                            RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                                .fill(Theme.Colors.card)
                                .overlay(
                                    RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                                        .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                                )
                        )
                }
                
                Button(action: isSignUp ? signup : login) {
                    if authManager.isLoading {
                        ProgressView()
                            .progressViewStyle(CircularProgressViewStyle(tint: Theme.Colors.onPrimary))
                    } else {
                        Text(isSignUp ? "Create Account" : "Sign In")
                    }
                }
                .font(Theme.Typography.label)
                .foregroundColor(Theme.Colors.onPrimary)
                .padding(.horizontal, Theme.Spacing.lg)
                .padding(.vertical, Theme.Spacing.md)
                .frame(maxWidth: .infinity)
                .background(
                    RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                        .fill(Theme.Colors.primary)
                        .opacity(authManager.isLoading || email.isEmpty || password.isEmpty ? 0.6 : 1.0)
                )
                .disabled(authManager.isLoading || email.isEmpty || password.isEmpty)
                
                Button(isSignUp ? "Already have an account? Sign In" : "Don't have an account? Sign Up") {
                    isSignUp.toggle()
                }
                .font(Theme.Typography.body)
                .foregroundColor(Theme.Colors.primary)
            }
            
            Spacer()
        }
        .padding(Theme.Spacing.xl)
        .background(Theme.Colors.surface)
        .navigationBarHidden(true)
    }
    
    private func login() {
        Task {
            await authManager.login(email: email, password: password)
        }
    }
    
    private func signup() {
        Task {
            await authManager.signup(email: email, password: password)
        }
    }
}

#Preview {
    WelcomeView()
}

#Preview {
    AuthSelectionView()
}

#Preview {
    EnhancedLoginView()
        .environmentObject(AuthManager())
}